.travis.yml

```diff
@@ -1,25 +1,37 @@
-sudo: false
-dist: trusty
+dist: xenial
 language: ruby
 bundler_args: --without documentation
-script: bundle exec rake compile test
-# Pin Rubygems to a working version. Sometimes it breaks upstream. Update now and then.
+
 before_install:
   - 'if [[ "$TRAVIS_OS_NAME" = "osx" ]]; then brew install openssl; fi'
-  - gem update --system 2.7.4
-  - gem update bundler
+  # rubygems 2.7.8 and greater include bundler
+  # remove 2.7.0 code when Travis removes rubygems 2.7.8 from ruby-head build
+  - |
+    rv="$(ruby -e 'STDOUT.write RUBY_VERSION')";
+    if   [ "$rv" \< "2.3" ]; then gem update --system 2.7.8 --no-document
+    elif [ "$rv" \< "2.7" ]; then gem update --system --no-document --conservative
+    elif [ "$rv" = "2.7.0" ]; then
+      lib="$(ruby -e 'STDOUT.write RbConfig::CONFIG["sitelibdir"]')";
+      rm -rf $lib/rubygems;
+      rm -f  $lib/rubygems.rb $lib/ubygems.rb;
+    fi
+
+script:
+  - bundle exec rake compile test
+
 env:
   global:
-    - TESTOPTS=-v
+    - TESTOPTS="-v --no-show-detail-immediately --order=random"
+
 rvm:
+  - ruby-head
   - 2.6
-  - 2.5
-  - 2.4
-  - 2.3
-  - 2.2
+  - 2.5.3
+  - 2.4.5
+  - 2.3.8
+  - 2.2.10
```

Comment:
I prefer to let these versions float to latest - did you find unexpected versions/behavior without pinning these specifically?

Reply:
Makes sense.  Will update.